flocker/common/test/test_version.py

```diff
@@ -146,269 +80,207 @@ def test_invalid_Version(self):
+        installable_version='0.3.2',
+        is_release=True,
+        is_weekly_release=False,
+        is_pre_release=False,
+    ),
+)
+WeeklyReleaseTests = build_version_test(
+    "WeeklyReleaseTests",
+    VersionCase(
+        version="0.3.2dev1",
+        flocker_version=FlockerVersion(
+            major="0",
+            minor="3",
+            micro="2",
+            weekly_release="1",
+        ),
+        doc_version='0.3.2dev1',
+        installable_version='0.3.2dev1',
+        is_release=False,
+        is_weekly_release=True,
+        is_pre_release=False,
+    ),
+)
+PreReleaseTests = build_version_test(
+    "PreReleaseTests",
+    VersionCase(
+        version='0.3.2pre1',
+        flocker_version=FlockerVersion(
+            major="0",
+            minor="3",
+            micro="2",
+            pre_release='1',
+        ),
+        doc_version="0.3.2pre1",
+        installable_version="0.3.2pre1",
+        is_release=False,
+        is_weekly_release=False,
+        is_pre_release=True,
+    ),
+)
+DevelopmentVersionTests = build_version_test(
+    "DevelopmentVersionTestss",
+    VersionCase(
+        version='0.3.2+1.gf661a6a',
+        flocker_version=FlockerVersion(
+            major="0",
+            minor="3",
+            micro="2",
+            commit_count='1',
+            commit_hash='f661a6a',
+        ),
+        doc_version="0.3.2+1.gf661a6a",
+        installable_version="0.3.2",
+        is_release=False,
+        is_weekly_release=False,
+        is_pre_release=False,
+    ),
+)
+DirtyVersionTests = build_version_test(
+    "DirtyVersionTests",
+    VersionCase(
+        version='0.3.2+1.gf661a6a.dirty',
+        flocker_version=FlockerVersion(
+            major="0",
+            minor="3",
+            micro="2",
+            commit_count='1',
+            commit_hash='f661a6a',
+            dirty='.dirty',
+        ),
+        doc_version="0.3.2+1.gf661a6a.dirty",
+        installable_version="0.3.2",
+        is_release=False,
+        is_weekly_release=False,
+        is_pre_release=False,
+    ),
+)
+DocReleaseTests = build_version_test(
+    "DocReleaseTests",
+    VersionCase(
+        version='0.3.2.post11',
+        flocker_version=FlockerVersion(
+            major="0",
+            minor="3",
+            micro="2",
+            documentation_revision='11',
+        ),
+        doc_version="0.3.2",
+        installable_version="0.3.2",
+        is_release=True,
+        is_weekly_release=False,
+        is_pre_release=False,
+    ),
+)
+DocReleaseDirtyTests = build_version_test(
+    "DocReleaseDirtyTests",
+    VersionCase(
+        version='0.3.2.post11+1.gf661a6a.dirty',
+        flocker_version=FlockerVersion(
+            major="0",
+            minor="3",
+            micro="2",
+            documentation_revision='11',
+            commit_count='1',
+            commit_hash='f661a6a',
+            dirty='.dirty',
+        ),
+        doc_version="0.3.2.post11+1.gf661a6a.dirty",
```

Comment:
These are explicitly bytes in the `build_version_test ` docstring, should they be explicitly bytes here?